init/src/boot_16.cpp

```diff
@@ -12,6 +12,8 @@
 #include "boot_32.hpp"
 
 #include "gdt_types.hpp"
+void __attribute__ ((noreturn)) rm_main();
+void __attribute__ ((noreturn)) foo(){ rm_main(); }
```

Comment:
Could you try to add the foo method below rm_main and avoid having to write rm_main twice?
And maybe add a comment as to why this necessary (You can just mention this fix a boot issue even without the reason).

Reply:
> Could you try to add the foo method below rm_main and avoid having to write rm_main twice?

What do you mean?

Also, I've added the docstrings above.

Reply:
I mean, can you move `void __attribute__ ((noreturn)) foo(){ rm_main(); }` below the definition of rm_main (at the very end of the file) and see if that works?

Reply:
Nope. Can't get to a shell.

Reply:
Thanks for testing it!